### PR TITLE
Use iPhone 13 Pro for iOS device tests

### PR DESCRIPTION
**Pull request by evil159**
### What does this pull request do?

Switches iOS device tests to use iPhone 13 Pro.

### What is the motivation and context behind this change?

Firebase recommendation.

### CODE REVIEW

**Review by kiryldz — APPROVED** [review]

**Review by kiryldz — COMMENTED** [review]
Interestingly tests fail now 😄

**Review by OdNairy — APPROVED** [review]

**Comment by OdNairy** [comment]
@evil159 there are no 15.8 for 13 pro on the firebase testlab
CleanShot 2024-03-01 at 11 30 21@2x

File: .circleci/config.yml [hunk 1]
```diff
@@ -143,7 +143,7 @@ jobs:
           command: |
             gcloud firebase test ios run \
               --test << parameters.workspace-path >>/build_products.zip \
-              --device model=iphone8,version=15.7 \
+              --device model=iphone13pro,version=15.8 \
```
Comment (OdNairy) [inline .circleci/config.yml:146]:
```suggestion
              --device model=iphone13pro,version=15.7 \
```

CleanShot 2024-03-01 at 11 30 21@2x

Comment (evil159) [reply .circleci/config.yml:146]: I guess I was too optimistic with the version availability, thanks for the suggestion 👍

**Pull request merged**